<a href="https://colab.research.google.com/github/ShenlyR/Volume-Lambung-Perahu/blob/main/Hitung_Volume_Lambung_Perahu.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Nama:** Shenly Lestari Riada

**NPM:** 25083010019

**Kelas:** A

**Link Github:** https://github.com/ShenlyR/Volume-Lambung-Perahu.git

**Import Library**

In [55]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from PIL import Image

Panjang Kapal
$$
L = 4\ m
$$

Pada sketsa perahu, satu interval atau satu kotak grid terdiri dari dua pola dash-kosong, maka $$
1\text{ interval}=2(0,1)=0,2\text{ m}
$$

Jumlah interval sepanjang kapal pada gambar sebesar 20 interval, sehingga
$$
L=20(0,2)=4\text{ m}
$$

Diperoleh, panjang kapal yang digunakan adalah 4 m

**Tampak Atas**

In [56]:
ukuran_dash_cm = 5
ukuran_celah_cm = 5

ukuran_dash_kosong_cm = ukuran_dash_cm + ukuran_celah_cm
ukuran_dash_kosong_m = ukuran_dash_kosong_cm / 100

# Satu kotak grid
dash_kosong_per_interval = 2

interval_m = (
    dash_kosong_per_interval *
    ukuran_dash_kosong_m
)

# Jumlah interval pada perahu
jumlah_interval = 20

# Panjang kapal pada perahu
panjang_kapal = jumlah_interval * interval_m

print("Ukuran 1 dash-kosong =", ukuran_dash_kosong_m, "m")
print("Ukuran 1 interval grid =", interval_m, "m")
print("Jumlah interval =", jumlah_interval)
print("Panjang kapal =", panjang_kapal, "m")

Ukuran 1 dash-kosong = 0.1 m
Ukuran 1 interval grid = 0.2 m
Jumlah interval = 20
Panjang kapal = 4.0 m


In [57]:
# interval
titik = np.arange(jumlah_interval + 1)

jarak = titik * interval_m

hasil_grid = pd.DataFrame({
    "Titik": titik,
    "Jarak (m)": jarak
})

hasil_grid["Jarak (m)"] = hasil_grid["Jarak (m)"].round(2)

hasil_grid

,Titik,Jarak (m)
0,0,0.0
1,1,0.2
2,2,0.4
3,3,0.6
4,4,0.8
5,5,1.0
6,6,1.2
7,7,1.4
8,8,1.6
9,9,1.8


**Tampak Samping**

In [58]:
jumlah_interval_samping = jumlah_interval
interval_samping_m = interval_m

print("Jumlah interval:", jumlah_interval_samping)
print("Jarak 1 interval:", interval_samping_m, "m")

Jumlah interval: 20
Jarak 1 interval: 0.2 m


In [59]:
panjang_samping = jumlah_interval_samping * interval_samping_m

print("Jumlah interval:", jumlah_interval_samping)
print("Panjang tampak samping:", panjang_samping, "m")

Jumlah interval: 20
Panjang tampak samping: 4.0 m


In [60]:
titik_samping = np.arange(jumlah_interval_samping + 1)
jarak_samping = titik_samping * interval_samping_m

hasil_tampak_samping = pd.DataFrame({
    "Titik": titik_samping,
    "Jarak(m)": jarak_samping
})

hasil_tampak_samping["Jarak(m)"] = hasil_tampak_samping["Jarak(m)"].round(2)
hasil_tampak_samping

,Titik,Jarak(m)
0,0,0.0
1,1,0.2
2,2,0.4
3,3,0.6
4,4,0.8
5,5,1.0
6,6,1.2
7,7,1.4
8,8,1.6
9,9,1.8


In [61]:
tinggi = np.array([
    0.15, 0.27, 0.36, 0.42, 0.47,
    0.50, 0.51, 0.52, 0.52, 0.52,
    0.52, 0.52, 0.52, 0.52, 0.51,
    0.48, 0.43, 0.38, 0.33, 0.24,
    0.01
])

print("Jumlah data tinggi =", len(tinggi))

Jumlah data tinggi = 21


**Hasil Tampak Atas dan Tampak Samping**

In [62]:
# Jumlah interval tampak atas
jumlah_interval_atas = jumlah_interval

titik_atas = np.arange(jumlah_interval_atas + 1)

# Jarak antar titik
interval_m_atas = interval_m

# Posisi setiap titik
jarak_atas = titik_atas * interval_m_atas

# Lebar lambung pada tampak atas
lebar_atas = np.array([
    0.45, 0.60, 0.70, 0.75, 0.80,
    0.80, 0.80, 0.80, 0.80, 0.80,
    0.80, 0.80, 0.80, 0.80, 0.80,
    0.75, 0.70, 0.60, 0.50, 0.35,
    0.15
])

# Lebar satu lambung
lebar_satu_lambung = lebar_atas / 2

# Tinggi dari tampak samping
tinggi_atas = tinggi

hasil_tinggi_atas = pd.DataFrame({
    "Titik": titik_atas,
    "Jarak (m)": jarak_atas,
    "Lebar satu lambung (m)": lebar_satu_lambung,
    "Tinggi (m)": tinggi_atas
})

hasil_tinggi_atas["Jarak (m)"] = hasil_tinggi_atas["Jarak (m)"].round(2)
hasil_tinggi_atas["Lebar satu lambung (m)"] = hasil_tinggi_atas["Lebar satu lambung (m)"].round(2)
hasil_tinggi_atas["Tinggi (m)"] = hasil_tinggi_atas["Tinggi (m)"].round(3)

hasil_tinggi_atas

,Titik,Jarak (m),Lebar satu lambung (m),Tinggi (m)
0,0,0.0,0.22,0.15
1,1,0.2,0.30,0.27
2,2,0.4,0.35,0.36
3,3,0.6,0.38,0.42
4,4,0.8,0.40,0.47
5,5,1.0,0.40,0.50
6,6,1.2,0.40,0.51
7,7,1.4,0.40,0.52
8,8,1.6,0.40,0.52
9,9,1.8,0.40,0.52


Rumus Luas Penampang:
$$
A_i = b_i \times h_i
$$

Keterangan:
- $A_i$ = luas penampang pada titik ke-$i$ ($m^2$)
- $b_i$ = lebar lambung pada titik ke-$i$ (m)
- $h_i$ = tinggi lambung pada titik ke-$i$ (m)

In [63]:
# Hitung luas penampang
hasil_tinggi_atas["Luas Penampang (m2)"] = (
    hasil_tinggi_atas["Lebar satu lambung (m)"] *
    hasil_tinggi_atas["Tinggi (m)"]
)

hasil_tinggi_atas["Luas Penampang (m2)"] = (
    hasil_tinggi_atas["Luas Penampang (m2)"].round(4)
)

hasil_tinggi_atas

,Titik,Jarak (m),Lebar satu lambung (m),Tinggi (m),Luas Penampang (m2)
0,0,0.0,0.22,0.15,0.0330
1,1,0.2,0.30,0.27,0.0810
2,2,0.4,0.35,0.36,0.1260
3,3,0.6,0.38,0.42,0.1596
4,4,0.8,0.40,0.47,0.1880
5,5,1.0,0.40,0.50,0.2000
6,6,1.2,0.40,0.51,0.2040
7,7,1.4,0.40,0.52,0.2080
8,8,1.6,0.40,0.52,0.2080
9,9,1.8,0.40,0.52,0.2080


In [64]:
print("Jumlah titik =", len(titik_atas))
print("Jumlah interval =", jumlah_interval_atas)
print("Jarak antar titik =", interval_m_atas, "m")
print("Simpson 1/3 terpenuhi? =", jumlah_interval_atas % 2 == 0)

assert len(titik_atas) == len(lebar_satu_lambung) == len(tinggi)
assert jumlah_interval_atas % 2 == 0

Jumlah titik = 21
Jumlah interval = 20
Jarak antar titik = 0.2 m
Simpson 1/3 terpenuhi? = True


**Hitung Volume Lambung Kapal**

**Aturan Simpson 1/3**

Rumus: $$
V = \frac{\Delta x}{3}
\left[
A_0 + A_n
+ 4(A_1 + A_3 + A_5 + \cdots + A_{n-1})
+ 2(A_2 + A_4 + A_6 + \cdots + A_{n-2})
\right]
$$

Keterangan:
- $V$ = volume satu lambung ($m^3$)
- $\Delta x$ = jarak antar titik (m)
- $A_0$ = luas penampang awal
- $A_n$ = luas penampang akhir
- Luas penampang dengan indeks ganjil dikalikan 4
- Luas penampang dengan indeks genap dikalikan 2

In [65]:
# data luas penampang
luas = hasil_tinggi_atas["Luas Penampang (m2)"].values

# aturan Simpson 1/3
volume_satu_lambung = (
    interval_m_atas / 3 *
    (
        luas[0]
        + luas[-1]
        + 4 * luas[1:-1:2].sum()
        + 2 * luas[2:-1:2].sum()
    )
)

volume_satu_lambung = round(volume_satu_lambung, 4)
print("Volume satu lambung =", volume_satu_lambung, "m3")

Volume satu lambung = 0.6427 m3


In [66]:
# Volume kedua lambung
volume_kedua_lambung = 2 * volume_satu_lambung

print(
    "Volume kedua lambung =",
    volume_kedua_lambung,
    "m3"
)

Volume kedua lambung = 1.2854 m3


**Aturan Trapesium**

Rumus: $$
V = \Delta x
\left[
\frac{A_0 + A_n}{2}
+ \sum_{i=1}^{n-1} A_i
\right]
$$

Keterangan:
- $V$ = volume satu lambung ($m^3$)
- $\Delta x$ = jarak antar titik atau interval (m)
- $A_0$ = luas penampang pada titik awal ($m^2$)
- $A_n$ = luas penampang pada titik akhir ($m^2$)
- $A_i$ = luas penampang pada titik ke-$i$ ($m^2$)
- $\sum_{i=1}^{n-1} A_i$ = jumlah seluruh luas penampang di antara titik awal dan titik akhir


In [67]:
# aturan trapesium
volume_trapesium = (
    interval_m_atas *
    (
        (luas[0] + luas[-1]) / 2
        + luas[1:-1].sum()
    )
)

print("Volume satu lambung - Trapesium =", volume_trapesium, "m3")

Volume satu lambung - Trapesium = 0.6416200000000001 m3


In [68]:
# volume kedua lambung
volume_kedua_trapesium = 2 * volume_trapesium

print("Volume kedua lambung - Trapesium =",volume_kedua_trapesium,"m3")

Volume kedua lambung - Trapesium = 1.2832400000000002 m3


**Hasil Perbandingan Antara kedua Aturan**

In [71]:
hasil_akhir = pd.DataFrame({
    "Aturan": [
        "Simpson 1/3",
        "Trapesium"
    ],
    "Volume 1 Lambung (m3)": [
        volume_satu_lambung,
        volume_trapesium
    ],
    "Volume 2 Lambung (m3)": [
        volume_kedua_lambung,
        volume_kedua_trapesium
    ]
})

hasil_akhir["Volume 1 Lambung (m3)"] = hasil_akhir["Volume 1 Lambung (m3)"].round(5)
hasil_akhir["Volume 2 Lambung (m3)"] = hasil_akhir["Volume 2 Lambung (m3)"].round(5)

# konveri ke liter
hasil_akhir["Volume 1 Lambung (liter)"] = (hasil_akhir["Volume 1 Lambung (m3)"] * 1000).round(2)
hasil_akhir["Volume 2 Lambung (liter)"] = (hasil_akhir["Volume 2 Lambung (m3)"] * 1000).round(2)

hasil_akhir

,Aturan,Volume 1 Lambung (m3),Volume 2 Lambung (m3),Volume 1 Lambung (liter),Volume 2 Lambung (liter)
0,Simpson 1/3,0.64270,1.28540,642.70,1285.40
1,Trapesium,0.64162,1.28324,641.62,1283.24


**Kesimpulan**

Berdasarkan hasil perhitungan, volume satu lambung perahu dengan metode Simpson 1/3 sebesar 0.64270 m³, sedangkan metode trapesium memberikan hasil sebesar 0.64162 m³.

Untuk kedua lambung, metode Simpson 1/3 menghasilkan volume sebesar 1.28540 m³, sedangkan metode trapesium menghasilkan volume sebesar 1.28324
 m³.

Hasil dari kedua metode cukup dekat, sehingga perbedaannya relatif kecil. Hasil perhitungan metode Simpson 1/3 menunjukkan nilai yang mendekati volume displacement pada satu lambung WL 0,4 m dalam jurnal sketsa perahu, yaitu sebesar 0.642 m³